tmc-plugin/src/fi/helsinki/cs/tmc/runners/ExerciseRunner.java

```diff
@@ -1,11 +1,16 @@
 package fi.helsinki.cs.tmc.runners;
 
+import com.google.common.base.Optional;
 import fi.helsinki.cs.tmc.data.TestRunResult;
 import fi.helsinki.cs.tmc.model.TmcProjectInfo;
 import java.util.concurrent.Callable;
 
+/**
+ * ExerciseRunner for implementing runner tasks
+ * Using Optional as a container for the result itself, if TestRunResult is not present
+ * compilation failure should be assumed.
```

Comment:
Not sure if the UX requires us to show the compile error in a dialog or anything. It should get logged to the output tab anyway.

But I don't object to having it, either in TestRunResult or as a special exception type (which would get rid of the Optional).